### PR TITLE
change 'elif' to 'case' when it is adequate on major scripts

### PR DESCRIPTION
**Pull request by nyxnor**
https://github.com/radio24/TorBox/pull/90 but part 2 on steroids.

### CODE REVIEW

**Comment by nyxnor** [comment]
If you ask if I have tested all, the answer is not yet, but I am trying to run all of the options :).

**Comment by radio24** [comment]
Ok, I will wait with the merge until you give me a green light regarding the testing.

**Comment by nyxnor** [comment]
I noticed that since the first PR.

I use two projects for that
* [editorconfig](https://editorconfig.org/) for tab identation and some more things
  * *py tab ident=4
  * *sh tab ident=2
* [shellcheck](https://github.com/koalaman/shellcheck) for proper syntax and tips | `sudo apt install -y shellcheck`
  * I use [this](https://github.com/nyxnor/onionservice/blob/main/.shellcheckrc)
* Nice CONTRIBUTING.md [from bash-completion](https://github.com/scop/bash-completion/blob/master/CONTRIBUTING.md)

**Comment by radio24** [comment]
In the mean while, I started with a contribution/coding guideline: https://github.com/radio24/TorBox/blob/master/CONTRIBUTING.md

**Comment by radio24** [comment]
In my test, I discovered that `grep "^PING_SERVER=" ${RUNFILE} | cut -c13-` returns http://google.com (this is what I wanted), but `grep "^PING_SERVER=" ${RUNFILE} | sed "s|.*http\?://|http\?://||g"` returns `google.com` (this is not what it should be). 

I will return to the old version because of the following reasons:
- If the user adds `http://something.else` or `https://something.else` in the `torbox.run`, I want that precisely this entry is used and not just `something.else`.
- There is a reason why some entries are with `http` instead of `https`. In a captive portal, only `http` request will bring up the login page, or dependent on the configuration, will find its way through the captive portal, but `https` will lead to an error.
- The above command goes under the category "reading settings from `torbox.run`". All these commands should be formulated in the same way, pulling `PING_SERVER` from `torbox.run` shouldn't be an exception.

**Comment by radio24** [comment]
@nyxnor If you received a notification from me that something with the new `PING_SERVER` routine would be wrong, then forget it - it was my mistake!

File: first_use [hunk 1]
```diff
@@ -141,137 +141,139 @@ finish_default_obfs4_step1()
...
-elif [ $FRESHINSTALLED = "FRESH_INSTALLED=2" ]; then
-	clear
-	sudo sed -i "s/^FRESH_INSTALLED=.*/FRESH_INSTALLED=3/" ${RUNFILE}
-	echo -e "${WHITE}[+] First time starting your TorBox!${NOCOLOR}"
-	echo -e "${WHITE}[+] Generating new unique SSH server keys for your TorBox installation!${NOCOLOR}"
-	(sudo rm -r /etc/ssh/ssh_host_*) 2> /dev/null
-	sudo ssh-keygen -A
-	echo ""
-	echo -e "${WHITE}[+] To take effect, you have to close and reopen all your SSH sessions.${NOCOLOR}"
-	FRESHINSTALLED="FRESH_INSTALLED=3"
```
Comment (radio24) [inline first_use:167]: The meaning of `FRESHINSTALLED="FRESH_INSTALLED=3"` is that after running this code block, the next (`elif [ $FRESHINSTALLED = "FRESH_INSTALLED=3" ]; then`) will be executed.

**Comment by nyxnor** [comment]
Ok, got time to review.
I was afraid it could fail because of a missing `;` or `;;` or `done` or `esac` at the end of case statements but shellcheck checks that also which is great. Not a green light yet.

**Comment by nyxnor** [comment]
Green light.

**Comment by radio24** [comment]
> Edit: shellcheck just saved me:
> 
> ```
> shellcheck -s bash -e SC2086,SC2164,SC3024 bridges_activate_old 
> 
> In bridges_activate_old line 78:
> if [ $number_configured_bridges_deactivated = 0 ]; then
> ^-- SC1046: Couldn't find 'fi' for this 'if'.
> ^-- SC1073: Couldn't parse this if expression. Fix to allow more checks.
> 
> 
> In bridges_activate_old line 217:
> 
> ^-- SC1047: Expected 'fi' matching previously mentioned 'if'.
> ^-- SC1072: Expected 'fi'. Fix any mentioned problems and try again.
> 
> For more information:
>   https://www.shellcheck.net/wiki/SC1046 -- Couldn't find 'fi' for this 'if'.
>   https://www.shellcheck.net/wiki/SC1047 -- Expected 'fi' matching previously...
>   https://www.shellcheck.net/wiki/SC1072 -- Expected 'fi'. Fix any mentioned ..
> ```

Actually, in both cases, `else` is not necessary and `fi` can be set after the `exit` command of the first part.

**Comment by radio24** [comment]
> Green light.

Not yet. There is one major problem in first_use: 
FRESHINSTALLED="FRESH_INSTALLED=3" will not have any effect and the next code block (case $FRESHINSTALLED in 3) will not be executed. :-(

Changing the parsed variable to execute the next part in the `if-elif-else` clause is something which worked with `elif`, but not with `case`.

**Comment by nyxnor** [comment]
Ok, seeing it now.

**Comment by nyxnor** [comment]
Normally on elifs I expected to just accept one of the statements, but as it read through all of them to execute multiples, any similar occurrence to this will fail if the loop with `while` as an example was not set. So if you remember any similar occurrences, let me know.

**Comment by radio24** [comment]
> Normally on elifs I expected to just accept one of the statements, but as it read through all of them to execute multiples, any similar occurrence to this will fail if the loop with `while` as an example was not set. So if you remember any similar occurrences, let me know.

Perfect solution - thanks!

**Pull request merged**